modules/core/client/services/menu.client.service.js

```diff
@@ -0,0 +1,196 @@
+      defaultRoles: ['user', 'admin'],
+      getMenu: getMenu,
+      menus: {},
+      removeMenu: removeMenu,
+      removeMenuItem: removeMenuItem,
+      removeSubMenuItem: removeSubMenuItem,
+      validateMenuExistance: validateMenuExistance
+    };
+
+    init();
+
+    return service;
+
+    // Add new menu object by menu id
+    function addMenu(menuId, options) {
+      options = options || {};
+
+      // Create the new menu
+      service.menus[menuId] = {
+        roles: options.roles || service.defaultRoles,
+        items: options.items || [],
+        shouldRender: shouldRender
+      };
+
+      // Return the menu object
+      return service.menus[menuId];
+    }
+
+    // Add menu item object
+    function addMenuItem(menuId, options) {
+      options = options || {};
+
+      // Validate that the menu exists
+      service.validateMenuExistance(menuId);
+
+      // Push new menu item
+      service.menus[menuId].items.push({
+        title: options.title || '',
+        state: options.state || '',
+        type: options.type || 'item',
+        class: options.class,
+        roles: ((options.roles === null || typeof options.roles === 'undefined') ? service.defaultRoles : options.roles),
+        position: options.position || 0,
+        items: [],
+        shouldRender: shouldRender
+      });
+
+      // Add submenu items
+      if (options.items) {
+        for (var i in options.items) {
+          if (options.items.hasOwnProperty(i)) {
+            service.addSubMenuItem(menuId, options.state, options.items[i]);
+          }
+        }
+      }
+
+      // Return the menu object
+      return service.menus[menuId];
+    }
+
+    // Add submenu item object
+    function addSubMenuItem(menuId, parentItemState, options) {
+      options = options || {};
+
+      // Validate that the menu exists
+      service.validateMenuExistance(menuId);
+
+      // Search for menu item
+      for (var itemIndex in service.menus[menuId].items) {
+        if (service.menus[menuId].items[itemIndex].state === parentItemState) {
+          // Push new submenu item
+          service.menus[menuId].items[itemIndex].items.push({
+            title: options.title || '',
+            state: options.state || '',
+            roles: ((options.roles === null || typeof options.roles === 'undefined') ? service.menus[menuId].items[itemIndex].roles : options.roles),
+            position: options.position || 0,
+            shouldRender: shouldRender
+          });
+        }
+      }
+
+      // Return the menu object
+      return service.menus[menuId];
+    }
+
+    // Get the menu object by menu id
+    function getMenu(menuId) {
+      // Validate that the menu exists
+      service.validateMenuExistance(menuId);
+
+      // Return the menu object
+      return service.menus[menuId];
+    }
+
+    function init() {
+      // A private function for rendering decision
+      shouldRender = function (user) {
+        if (!!~this.roles.indexOf('*')) {
```

Comment:
it will work, but not sure what this change has to do with angular style guide?  this is not throwing any exceptions, and this is not logic that I introduced. https://github.com/meanjs/mean/blob/master/modules/core/client/services/menus.client.service.js#L14